### DISCUSSION TITLE
Add option to make layer invisible for XrayDyn

### DISCUSSION THREAD

**Discussion by jpudell** [dschick/udkm1Dsim#145] [Ideas]
The simulation of a sample consisting of amorphous and crystalline layers works quite well for the heat and strain simulation.
But to get get the diffraction pattern of this sample is a problem: all layers must be defined as crystalline, but this leads to Bragg peaks of the "amorphous layer" which interfere with the signal of the crystalline layers. To suppress this, an option is to replace the atom with light atom like He; but this again leads to problems in the strain simulation.
I propose to add an option to set the structure factor for this layer to 0.

**Comment by dschick** [comment]
Hi @jpudell 

I was actually also thinking a bit about that issue when starting to include also `AmorphousLayer`s.
In general, the `XrayDyn` formalism cannot handle `AmorphousLayer`s but the `XrayDynMag` formalism can, to some extent, handle crystalline media (`UnitCell`s).
E.g. you will also see "Bragg peaks" when stacking 5 Å `AmorphousLayer`s of two different materials to imitate a `UnitCell` when using the `XrayDynMag` formalism. However, this will not work when both layers are the same, like elemental Ni.
When neglecting the magnetic scattering, the `XrayDynMag` formalism is similar to Parrat's method (which is also on the agenda to be implemented #65 )

If you like to stick with the `XrayDyn` formalism, you can also think of increasing the Debye-Waller factor (`deb_wal_fac`) of the according layers.
Alternatively, as you suggested, changing the atomic scattering factors of an atom can be achieved by manipulating the `atomic_form_factor_coeff` attribute of the according `Atom`s. This attribute holds the tabulated Henke or Chantler data (E, f1, f2) and can be simply overwritten.

Best

Daniel
